Storage/SQL/SQLiteHistoryRecommendations.swift

```diff
@@ -171,36 +174,35 @@ extension SQLiteHistory: HistoryRecommendations {
         return [(clearHighlightsQuery, nil), (sql, args)]
     }
 
-    // Checks if there are more than 100k items in the `visits` table.
+    // Checks if there are more than 250k items in the `history` table.
     // This is used as an indicator that we should clean up old history.
     func checkIfCleanupIsNeeded() -> Deferred<Maybe<Bool>> {
-        let sql = "SELECT COUNT(*) FROM \(TableVisits)"
+        let sql = "SELECT COUNT(*) FROM \(TableHistory)"
         return self.db.runQuery(sql, args: nil, factory: IntFactory) >>== { cursor in
-            guard let visitCount = cursor[0], visitCount > 100000 else {
+            guard let historyCount = cursor[0], historyCount > SQLiteHistory.MaxHistoryRowCount else {
```

Comment:
Consider altering the query to be `SELECT COUNT(rowid) > \(SQLiteHistory.MaxHistoryRowCount)`, then use a bool-ish comparison. In theory that can stop counting once it hits the larger number!